File: project/core/module/rest-service/src/main/java/vdi/service/rest/conversion/BioprojectIDReference.java
package vdi.service.rest.conversion;

import org.jetbrains.annotations.NotNull;
import vdi.service.rest.generated.model.BioprojectIDReferenceImpl;

public class BioprojectIDReference extends BioprojectIDReferenceImpl {
  public BioprojectIDReference(@NotNull vdi.model.meta.BioprojectIDReference reference) {
    setId(reference.getId());
    setDescription(reference.getDescription());
  }
}

File: project/core/module/rest-service/src/main/kotlin/vdi/service/rest/server/conversion/BioprojectIDReference.kt
package vdi.service.rest.server.conversion

import vdi.service.rest.generated.model.BioprojectIDReferenceImpl
import vdi.model.meta.BioprojectIDReference as InternalType
import vdi.service.rest.generated.model.BioprojectIDReference as RamlType

internal fun BioprojectIDReference(ref: InternalType): RamlType =
  BioprojectIDReferenceImpl().apply {
    id = ref.id
    description = ref.description
  }